### ISSUE TITLE
Use the offcial template of Qwen2-7B-Instruct-AWQ failed 

### ISSUE THREAD

**Issue by oyy2000** [QwenLM/Qwen2#547]
## Codes
The offcial template of using "Qwen/Qwen2-7B-Instruct-AWQ".

from transformers import AutoModelForCausalLM, AutoTokenizer
device = "cuda" # the device to load the model onto

model = AutoModelForCausalLM.from_pretrained(
    "Qwen/Qwen2-7B-Instruct-AWQ",
    torch_dtype="auto",
    device_map="auto"
)
tokenizer = AutoTokenizer.from_pretrained("Qwen/Qwen2-7B-Instruct-AWQ")

prompt = "Give me a short introduction to large language model."
messages = [
    {"role": "system", "content": "You are a helpful assistant."},
    {"role": "user", "content": prompt}
]
text = tokenizer.apply_chat_template(
    messages,
    tokenize=False,
    add_generation_prompt=True
)
model_inputs = tokenizer([text], return_tensors="pt").to(device)

generated_ids = model.generate(
    model_inputs.input_ids,
    max_new_tokens=512
)
generated_ids = [
    output_ids[len(input_ids):] for input_ids, output_ids in zip(model_inputs.input_ids, generated_ids)
]

response = tokenizer.batch_decode(generated_ids, skip_special_tokens=True)[0]

## Errors
[2827](https://vscode-remote+ssh-002dremote-002bterminator8-002ecs-002erice-002eedu.vscode-resource.vscode-cdn.net/scratch/zx22/zijie/anaconda/envs/autoawq_ouyangyang/lib/python3.9/site-packages/transformers/generation/utils.py:2827) # sample
   [2828](https://vscode-remote+ssh-002dremote-002bterminator8-002ecs-002erice-002eedu.vscode-resource.vscode-cdn.net/scratch/zx22/zijie/anaconda/envs/autoawq_ouyangyang/lib/python3.9/site-packages/transformers/generation/utils.py:2828) probs = nn.functional.softmax(next_token_scores, dim=-1)
-> [2829](https://vscode-remote+ssh-002dremote-002bterminator8-002ecs-002erice-002eedu.vscode-resource.vscode-cdn.net/scratch/zx22/zijie/anaconda/envs/autoawq_ouyangyang/lib/python3.9/site-packages/transformers/generation/utils.py:2829) next_tokens = torch.multinomial(probs, num_samples=1).squeeze(1)
   [2831](https://vscode-remote+ssh-002dremote-002bterminator8-002ecs-002erice-002eedu.vscode-resource.vscode-cdn.net/scratch/zx22/zijie/anaconda/envs/autoawq_ouyangyang/lib/python3.9/site-packages/transformers/generation/utils.py:2831) # finished sentences should have their next token be a padding token
   [2832](https://vscode-remote+ssh-002dremote-002bterminator8-002ecs-002erice-002eedu.vscode-resource.vscode-cdn.net/scratch/zx22/zijie/anaconda/envs/autoawq_ouyangyang/lib/python3.9/site-packages/transformers/generation/utils.py:2832) if eos_token_id is not None:

RuntimeError: probability tensor contains either `inf`, `nan` or element < 0

## Environment:
Package                       Version
----------------------------- --------------
accelerate                    0.30.1
aiohttp                       3.9.5
aiosignal                     1.3.1
annotated-types               0.7.0
anyio                         4.4.0
argon2-cffi                   23.1.0
argon2-cffi-bindings          21.2.0
arrow                         1.3.0
asttokens                     2.4.1
async-lru                     2.0.4
async-timeout                 4.0.3
attrs                         23.2.0
autoawq                       0.2.4
autoawq_kernels               0.0.6
Babel                         2.15.0
beautifulsoup4                4.12.3
bleach                        6.1.0
certifi                       2024.6.2
cffi                          1.16.0
charset-normalizer            3.3.2
cmake                         3.25.0
comm                          0.2.2
datasets                      2.19.2
debugpy                       1.8.1
decorator                     5.1.1
deepspeed                     0.14.2
defusedxml                    0.7.1
dill                          0.3.8
einops                        0.8.0
exceptiongroup                1.2.1
executing                     2.0.1
fastjsonschema                2.19.1
filelock                      3.14.0
fqdn                          1.5.1
frozenlist                    1.4.1
fsspec                        2024.3.1
h11                           0.14.0
hjson                         3.1.0
httpcore                      1.0.5
httpx                         0.27.0
huggingface-hub               0.23.3
idna                          3.7
importlib_metadata            7.1.0
ipykernel                     6.29.4
ipython                       8.18.1
ipywidgets                    8.1.3
isoduration                   20.11.0
jedi                          0.19.1
Jinja2                        3.1.4
json5                         0.9.25
jsonpointer                   2.4
jsonschema                    4.22.0
jsonschema-specifications     2023.12.1
jupyter                       1.0.0
jupyter_client                8.6.2
jupyter-console               6.6.3
jupyter_core                  5.7.2
jupyter-events                0.10.0
jupyter-lsp                   2.2.5
jupyter_server                2.14.1
jupyter_server_terminals      0.5.3
jupyterlab                    4.2.1
jupyterlab_pygments           0.3.0
jupyterlab_server             2.27.2
jupyterlab_widgets            3.0.11
lit                           15.0.7
MarkupSafe                    2.1.5
matplotlib-inline             0.1.7
mistune                       3.0.2
mpmath                        1.3.0
multidict                     6.0.5
multiprocess                  0.70.16
nbclient                      0.10.0
nbconvert                     7.16.4
nbformat                      5.10.4
nest-asyncio                  1.6.0
networkx                      3.2.1
ninja                         1.11.1.1
notebook                      7.2.0
notebook_shim                 0.2.4
numpy                         1.26.4
nvidia-cublas-cu12            12.1.3.1
nvidia-cuda-cupti-cu12        12.1.105
nvidia-cuda-nvrtc-cu12        12.1.105
nvidia-cuda-runtime-cu12      12.1.105
nvidia-cudnn-cu12             8.9.2.26
nvidia-cufft-cu12             11.0.2.54
nvidia-curand-cu12            10.3.2.106
nvidia-cusolver-cu12          11.4.5.107
nvidia-cusparse-cu12          12.1.0.106
nvidia-nccl-cu12              2.20.5
nvidia-nvjitlink-cu12         12.5.40
nvidia-nvtx-cu12              12.1.105
overrides                     7.7.0
packaging                     24.0
pandas                        2.2.2
pandocfilters                 1.5.1
parso                         0.8.4
peft                          0.11.1
pexpect                       4.9.0
pillow                        10.2.0
pip                           24.0
platformdirs                  4.2.2
prometheus_client             0.20.0
prompt_toolkit                3.0.46
psutil                        5.9.8
ptyprocess                    0.7.0
pure-eval                     0.2.2
py-cpuinfo                    9.0.0
pyarrow                       16.1.0
pyarrow-hotfix                0.6
pycparser                     2.22
pydantic                      2.7.3
pydantic_core                 2.18.4
Pygments                      2.18.0
pynvml                        11.5.0
python-dateutil               2.9.0.post0
python-json-logger            2.0.7
pytz                          2024.1
PyYAML                        6.0.1
pyzmq                         26.0.3
qtconsole                     5.5.2
QtPy                          2.4.1
referencing                   0.35.1
regex                         2024.5.15
requests                      2.32.3
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rpds-py                       0.18.1
safetensors                   0.4.3
scipy                         1.13.1
Send2Trash                    1.8.3
setuptools                    69.5.1
six                           1.16.0
sniffio                       1.3.1
soupsieve                     2.5
stack-data                    0.6.3
sympy                         1.12.1
terminado                     0.18.1
tiktoken                      0.7.0
tinycss2                      1.3.0
tokenizers                    0.19.1
tomli                         2.0.1
torch                         2.0.1+cu118
torchaudio                    2.0.2+cu118
torchvision                   0.15.2+cu118
tornado                       6.4
tqdm                          4.66.4
traitlets                     5.14.3
transformers                  4.40.0
transformers-stream-generator 0.0.4
triton                        2.0.0
types-python-dateutil         2.9.0.20240316
typing_extensions             4.12.1
tzdata                        2024.1
uri-template                  1.3.0
urllib3                       2.2.1
wcwidth                       0.2.13
webcolors                     1.13
webencodings                  0.5.1
websocket-client              1.8.0
wheel                         0.43.0
widgetsnbextension            4.0.11
xxhash                        3.4.1
yarl                          1.9.4
zipp                          3.19.2
zstandard                     0.22.0

**Comment by achaosss** [comment]
I have another question.
  File "C:\Users\Z19070948\AppData\Local\anaconda3\envs\py311\Lib\site-packages\transformers\quantizers\quantizer_awq.py", line 78, in _process_model_before_weight_loading
    from ..integrations import get_keys_to_not_convert, replace_quantization_scales, replace_with_awq_linear
  File "<frozen importlib._bootstrap>", line 1229, in _handle_fromlist
  File "C:\Users\Z19070948\AppData\Local\anaconda3\envs\py311\Lib\site-packages\transformers\utils\import_utils.py", line 1525, in __getattr__        
    module = self._get_module(self._class_to_module[name])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Z19070948\AppData\Local\anaconda3\envs\py311\Lib\site-packages\transformers\utils\import_utils.py", line 1537, in _get_module        
    raise RuntimeError(
RuntimeError: Failed to import transformers.integrations.bitsandbytes because of the following error (look up to see its traceback):
name 'torch' is not defined

**Comment by jklj077** [comment]
@achaosss there is a bug in transformers (<https://www.github.com/huggingface/transformers/pull/31243>). you can install from the latest source which includes the PR that fix the issue, or install `transformers<4.41.0`.

**Comment by jklj077** [comment]
Hi,

The model is working properly in my environment with the following package versions:
```
accelerate             0.30.1
autoawq                0.2.5
autoawq_kernels        0.0.6
optimum                1.20.0
torch                  2.2.1+cu121
transformers           4.40.2
```
and NVIDIA driver 535.161.08 and 1 or 4 or 8 NVIDIA A10 GPUs.

In addition, I noticed that you had installed `torch` that is compiled for cu118 and `autoawq_kernels`` which by default is compiled for cu121. If that's the case, there could be a compatibility problem. Please try install `torch` compiled for cu121 or `autoawq_kernels` compiled for cu118, following their official instructions.

If you're running on multiple GPUs, there are also other possibilities. For example, there were reports that the model works on a single GPU but not on multiple GPUs across PCI-E switches, which appears to be related to the hardware configuration and the NVIDIA driver version.

**Comment by oyy2000** [comment]
Thank you soooo much! I will try!

**Comment by achaosss** [comment]
> @achaosss there is a bug in transformers (https://www.github.com/huggingface/transformers/pull/31243). you can install from the latest source which includes the PR that fix the issue, or install `transformers<4.41.0`.
> 
> If you have other questions, please open a new issue. I'm marking this as off topic.

Thanks you I will try it
